src/hostname/hostnamed.c

```diff
@@ -546,6 +553,34 @@ static bool valid_deployment(const char *deployment) {
         return in_charset(deployment, VALID_DEPLOYMENT_CHARS);
 }
 
+/* fallback_ansi_color() hashes a color automatically from the hostname if none was
+ * explicitly set */
+static char* fallback_ansi_color(Context *c) {
+        uint64_t hash;
+        static const uint8_t key[16] = { 0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+                                         0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f};
+        double h, s = 90, v = 70;
+        uint8_t r, g, b;
+        const char* hostname;
+        _cleanup_free_ char *color = NULL;
+        int k;
+
+        assert(c);
+
+        hostname = isempty(c->data[PROP_STATIC_HOSTNAME_SUBSTITUTED_WILDCARDS]) ?
+                "localhost" : c->data[PROP_STATIC_HOSTNAME_SUBSTITUTED_WILDCARDS];
+        hash = siphash24_string(hostname, key);
+        h = ((double) (hash >> 21) * 360.0 / 8796093022208.0);
+        hsv_to_rgb(h, s, v, &r, &g, &b);
+        k = asprintf(&color, "\x1B[0;38;2;%u;%u;%um", r, g, b);
```

Comment:
please drop the `\x1B[0;` prefix and the `m` suffix. and let this be prefixed/appended client-side